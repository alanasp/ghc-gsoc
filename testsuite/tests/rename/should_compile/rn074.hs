{-# OPTIONS_GHC -fwarn-warnings-deprecations #-}

-- Test deprecation of variable and data constructor exports

module ShouldCompile where

import Rn074_B(funcRn074_C) --non-deprecated
import Rn074_C(funcRn074_C) --deprecated at definition
import Rn074_A(funcRn074_C) --non-deprecated

main = funcRn074_C --warning because deprecated at the defining module
