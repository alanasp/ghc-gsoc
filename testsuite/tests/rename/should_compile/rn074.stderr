rn074.hs:7:1: warning: [-Wdeprecations (in -Wdefault)]
    Deprecated: "Rn073_C shall be imported directly"

rn074.hs:8:1: warning: [-Wdeprecations (in -Wdefault)]
    Deprecated: "funcRn074_C is deprecated"

rn074.hs:9:1: warning: [-Wdeprecations (in -Wdefault)]
    Deprecated: "Rn073_C shall be imported directly"

rn074.hs:11:8: warning: [-Wdeprecations (in -Wdefault)]
    In the use of ‘funcRn074_C’
    (imported from Rn074_A, but defined in Rn074_C):
    Deprecated: "Rn073_C shall be imported directly"
